pkg/gofr/datasource/sql/sql.go

```diff
@@ -33,13 +33,20 @@ type DBConfig struct {
 }
 
 func NewSQL(configs config.Config, logger datasource.Logger, metrics Metrics) *DB {
+	logger.Debugf("verifying database configurations from config file")
+
 	dbConfig := getDBConfig(configs)
 
 	// if Hostname is not provided, we won't try to connect to DB
 	if dbConfig.Dialect != sqlite && dbConfig.HostName == "" {
+		logger.Debugf("not connecting to database as database configurations aren't available")
 		return nil
 	}
 
+	logger.Debugf("generating database connection string for '%s'", dbConfig.Dialect)
+
+	logger.Debugf("generating database connection string for '%s'", dbConfig.Dialect)
```

Comment:
this log is added twice, we should remove one

Reply:
This looks like unnecessarily verbose logging. All of these log lines can be combined on the line 50 itself.